src/main/java/teammates/common/datatransfer/questions/FeedbackMcqQuestionDetails.java

```diff
@@ -95,19 +116,75 @@ public boolean extractQuestionDetails(
                 }
             }
 
-            setMcqQuestionDetails(numOfMcqChoices, mcqChoices, mcqOtherEnabled);
+            String hasAssignedWeightsString = HttpRequestHelper.getValueFromParamMap(
+                    requestParameters, Const.ParamsNames.FEEDBACK_QUESTION_MCQ_WEIGHTS_ASSIGNED);
+            boolean hasAssignedWeights = "on".equals(hasAssignedWeightsString);
+            List<Double> mcqWeights = getMcqWeights(
+                    requestParameters, numMcqChoicesCreated, hasAssignedWeights);
+            double mcqOtherWeight = getMcqOtherWeight(requestParameters, mcqOtherEnabled, hasAssignedWeights);
+            setMcqQuestionDetails(
+                    numOfMcqChoices, mcqChoices, mcqOtherEnabled, hasAssignedWeights, mcqWeights, mcqOtherWeight);
         } else {
             setMcqQuestionDetails(FeedbackParticipantType.valueOf(generatedMcqOptions));
         }
         return true;
     }
 
-    private void setMcqQuestionDetails(int numOfMcqChoices,
-            List<String> mcqChoices,
-            boolean otherEnabled) {
+    private List<Double> getMcqWeights(Map<String, String[]> requestParameters,
+            int numMcqChoicesCreated, boolean hasAssignedWeights) {
+        List<Double> mcqWeights = new ArrayList<>();
+
+        if (!hasAssignedWeights) {
+            return mcqWeights;
+        }
+
+        for (int i = 0; i < numMcqChoicesCreated; i++) {
+            String choice = HttpRequestHelper.getValueFromParamMap(
+                    requestParameters, Const.ParamsNames.FEEDBACK_QUESTION_MCQCHOICE + "-" + i);
+            String weight = HttpRequestHelper.getValueFromParamMap(
+                    requestParameters, Const.ParamsNames.FEEDBACK_QUESTION_MCQ_WEIGHT + "-" + i);
+
+            if (choice != null && !choice.trim().isEmpty() && weight != null) {
+                try {
+                    // Do not add weight to mcqWeights if the weight cannot be parsed
+                    mcqWeights.add(Double.parseDouble(weight));
+                } catch (NumberFormatException e) {
+                    log.warning("Failed to parse weight for MCQ question: " + weight);
+                }
+            }
+        }
+
+        return mcqWeights;
+    }
+
+    private double getMcqOtherWeight(Map<String, String[]> requestParameters,
+            boolean mcqOtherEnabled, boolean hasAssignedWeights) {
+
+        double mcqOtherWeight = this.mcqOtherWeight;
```

Comment:
In case if the user tries to edit the question, and leaves the `otherWeight` field empty, then if I set `double mcqOtherWeight = 0` here, the existing otherWeight will be changed to zero, instead of preserving the previous weight.

Reply:
Yes. That's is the limitation of our update logic for question detail. However, I would say in the normal condition, we would always have the old `mcqOtherWeight` submitted in the UI.  Let's have this assumption until we have a better way to modify question details.

Reply:
Okay. Will use `0` here.